src/frontend/js/components/PaymentButton/index.tsx

```diff
@@ -252,8 +283,39 @@ const PaymentButton = ({ product, billingAddress, creditCard, onSuccess }: Payme
     }
   }, [state]);
 
+  const api = useJoanieApi();
+
+  const openContract = async (e: React.MouseEvent) => {
+    if (!product.contract_definition) {
+      return;
+    }
+    e.stopPropagation();
+    e.preventDefault();
+    const blob = await api.contractDefinitions.previewTemplate(product.contract_definition.id);
+    // eslint-disable-next-line compat/compat
+    const file = window.URL.createObjectURL(blob);
+    window.open(file);
+  };
+
+  const label = (
+    // eslint-disable-next-line jsx-a11y/anchor-is-valid
+    <a href="#" onClick={openContract}>
+      {intl.formatMessage(messages.termsMessage)}
+    </a>
+  );
```

Comment:
I feel this is not a good practice to nest interactive element. Indeed, the user has to click on this label to check the box but if it clicks on the second part it will also download the preview that could be surprising and annoying.

Reply:
If the user click of the red part the checkbox is not checked. I often see this pattern, however, do have any other way of doing in mind?

<img width="296" alt="Capture d’écran 2023-11-24 à 14 16 07" src="https://github.com/openfun/richie/assets/9628870/9fcec5aa-85e9-40ac-a111-25fb4bf180c9">

Reply:
Yep, I turned my though and indeed this is a common pattern. I would just suggest a little change in order to increase the click area to check the box.

What do you think to modify the label to :
> By checking this box, you accept the General Terms of Sale